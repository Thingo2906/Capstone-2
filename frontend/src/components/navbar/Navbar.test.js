import React from "react";
import Navbar from "./Navbar";
import UserContext from "../../auth/UserContext";
import {render} from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { hasUncaughtExceptionCaptureCallback } from "process";
mockLogout = jest.fn();

describe("Navbar Component", function(){
    test("render without crashing", function(){
        const currentUser ={};
        render(
        <BrowserRouter> 
        <UserContext.Provider value={{ currentUser }}>
          <Navbar logout={mockLogout} />
        </UserContext.Provider>
        </BrowserRouter> )        
    });
    test("matches Snapshot", function(){
        const currentUser = {};
        const { asFragment } = render(
          <BrowserRouter>
            <UserContext.Provider value={{ currentUser }}>
              <Navbar logout={mockLogout} />
            </UserContext.Provider>
          </BrowserRouter>
        ); 
        expect(asFragment()).toMatchSnapshot();
    })
    test("renders Navbar when user is logged in", function(){
        const currentUser = {username: "testuser"};
        const {getByText} = render(
          <BrowserRouter>
            <UserContext.Provider value={{ currentUser }}>
              <Navbar logout={mockLogout} />
            </UserContext.Provider>
          </BrowserRouter>
        ); 
        expect(getByText("Movies")).toBeInTheDocument();
        expect(getByText("Profile")).toBeInTheDocument();
        expect(getByText("MyList")).toBeInTheDocument();
        expect(getByText("Search")).toBeInTheDocument();
    });

  test('renders Navbar when user is logged out', () => {
    const currentUser = null;
    const {getByText} = render(
         <BrowserRouter>
            <UserContext.Provider value={{ currentUser }}>
              <Navbar logout={mockLogout} />
            </UserContext.Provider>
          </BrowserRouter>
      
    );
    expect(getByText("Login")).toBeInTheDocument();
    expect(getByText("Sign Up")).toBeInTheDocument();
  });
    
})