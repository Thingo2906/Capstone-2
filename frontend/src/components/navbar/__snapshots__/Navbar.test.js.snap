// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component matches Snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      href="/homepage"
    >
      <img
        alt="Movie Logo"
        class="nav-logo"
        src="../logo-png.png"
      />
    </a>
    <div
      class="navbar-collapse"
    >
      <div
        class="logged-in-nav"
      >
        <div
          class="nav"
        >
          <div
            class="navbar-links-left"
          >
            <ul
              class="navbar-links"
            >
              <li>
                <a
                  class="nav-link"
                  href="/movies"
                >
                  Movies
                </a>
              </li>
              <li>
                <a
                  class="nav-link"
                  href="/profile"
                >
                  Profile
                </a>
              </li>
              <li>
                <a
                  class="nav-link"
                  href="/mylist"
                >
                  MyList
                </a>
              </li>
              <li>
                <a
                  class="nav-link"
                  href="/search"
                >
                  Search
                </a>
              </li>
            </ul>
          </div>
          <div
            class="logout"
          >
            <ul>
              <li>
                <a
                  class="nav-logout"
                  href="/homepage"
                >
                  Log out 
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
